assignment_1/1_permutations.py

```diff
@@ -0,0 +1,44 @@
+#Given two strings, write a method to decide if one is a permutation of the other
+# Magdalena Grodzińska
+
+import re
+from collections import Counter
+
+# helper funtions
+
+def validate_input(input):
+    pattern = '[a-zA-Z]+'
+
+    result = re.fullmatch(pattern, input)
+
+    if result == None:
+        return False
+    else:
+        return True
+
+def check_letters_match(text1, text2):
+    res1 = Counter(text1)
+    res2 = Counter(text2)
+
+    return res1 == res2
+
+# main function
+
+def check_permutation(text1, text2):
+    if validate_input(text1) and validate_input(text2):
```

Comment:
I see - if this decision is based on requirements (and it is indeed fair to consider answers on slack being requirements) than that's fine.

If such a situation arose professionally (or in an interview) then it's worth mentioning and documenting (for example, as a comment) the interesting requirements and their implications - in this case performance implications - so that for example the next person (or you in 3 months) refactoring the code would still be aware of them.